ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStorageHelixParticipant.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.account.AccountService;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import org.apache.helix.participant.StateMachineEngine;
+
+
+/**
+ * Helix participant
+ */
+public class AzureStorageHelixParticipant extends HelixVcrClusterParticipant {
```

Comment:
For the Helix Participant, do we need do any change?
What's the problem with the existing HelixVcrClusterParticipant?

Reply:
Readability and Operability.

There are several problems with using HelixVcrClusterParticipant for V2. 

1. Creating new counterpart classes enables switching from V1 to V2 and back easily, using _exactly_ one config variable which is azureBackupVersion. Otherwise, there are 4-5 config variables you'll have to remember and change in order to switch. cloudContainerCompactionEnabled is one among those configs when we use V1 helix participant. Furthermore, V2 will first be deployed in a single cluster while others remain on V1. Additionally, switching a V2 cluster back to V1 in case of any issues would be a no-brainer if there is exactly one config. A single config minimizes operator errors during such deployments.

2. Secondly, registerContainerDeletionSyncTask needs to be over-written. The backup format changes from V1 to V2 and V1 clean-up tasks will simply not work. Over-writing it completely is cleaner than meddling with existing implementation.

3. Finally, creating new classes does not harm performance, reliability or availability. It doesn't even harm readability in this case. So if there are other concerns, I don't understand them.

Reply:
Item2 is a good reason for me. Yes, we can use the new class.

But,
item 1: If we use template and provide the corresponding <T>, it can solve problem 1.
item 3: Although creating new class won't harm performance, we don't help with readability since have more code to deal with.

Reply:
got it, understood.